### ISSUE TITLE
Only audio and QuickTime Splash Screen instead of video

### ISSUE THREAD

**Issue by fidoriel** [streamyfin/streamyfin#67]
**Describe the bug**
Sometimes only audio is played. Video Shows only QuickTime symbol.
I will investigate server logs to give a more precise error log.

**To Reproduce**
1. Server with disabled transcoding. MKV file with h265.
2.  Play this movie. Only audio plays, video shows QuickTime symbol. Video played in previous version fine.

**Smartphone:**
 - Device: iPhone13Mini
 - OS: iOS17.5.1
 - Version: 0.6.1

**Comment by fredrikburmester** [comment]
This is because the client does not support that file format/client. Since the server has transcoding turned off you can't play the file on the client. This is a limitation of apple/android devices and the default supported codecs.

**Comment by fidoriel** [comment]
That is expected behavior for formats that are not supported out of the box. But imho h265 is supported out of the box on ios. With a previous version of streamyfin the same movie worked fine. Some release last week broke it.

**Video**
- **Title:** 1080p HEVC SDR
- **Codec:** HEVC
- **AVC:** No
- **Profile:** Main 10
- **Level:** 120
- **Resolution:** 1920x1036
- **Aspect Ratio:** 1.85:1
- **Anamorphic:** No
- **Interlaced:** No
- **Framerate:** 23.976025
- **Bitrate:** 10783 kbps
- **Bit Depth:** 10 bit
- **Video Range:** SDR
- **Video Dynamic Range:** SDR
- **Color Space:** bt709
- **Color Transfer:** bt709
- **Primaries:** bt709
- **Pixel Format:** yuv420p10le
- **Reference Frames:** 1

**Comment by fredrikburmester** [comment]
What's the container? Mkv?

**Comment by fidoriel** [comment]
Yes, MKV Container. But it should remux it? Audio works.

**Comment by fredrikburmester** [comment]
I don't know how Jellyfin works but I don't think it can remux if transcoding is turned off on the server. We could maybe build remuxing into the app from mkv to mp4 container (?). If that's possible.

**Comment by michi7801** [comment]
I'm having the same issue with h265 playback on the iOS app. Except I'm using a mp4 container. I have not tested playback with previous of the app versions though.

It works fine on the official app and h265 is definitely supported on my iPhone 12...

**Comment by fredrikburmester** [comment]
That's interesting! Could you send the information about that file? @michi7801

**Comment by michi7801** [comment]
Sure thing:

Title: 1080p HEVC SDR
Codec: HEVC
AVC: No
Profile: Main
Level: 120
Resolution: 1920x1080
Aspect Ratio: 16:9
Anamorphic: No
Interlaced: No
Framerate: 25
Bitrate: 1429 kbps
Bit Depth: 8 bit
Video Range: SDR
Video Dynamic Range: SDR
Color Space: bt709
Color Transfer: bt709
Primaries: bt709
Pixel Format: yuv420p
Reference Frames: 1

**Comment by fredrikburmester** [comment]
That file seems quite small, would you mind sharing it with me? Send me an email and i'll send you an upload link. fredrik.burmester@gmail.com @michi7801

**Comment by fredrikburmester** [comment]
I've looked into this, it seems that iOS devices do not support MPEG-H Part 2/HEVC, not even on my iPhone 15 Pro. This is causing the video playback to fail. A solution for this is to switch to the "Old" profile in the settings, forcing the transcoding from HEVC->H264. 

Now, why it works in other apps but not Streamyfin is probably because of the Device Profile we're using, even though we're using the same as the official Jellyfin app...

**Comment by fidoriel** [comment]
hevc/h265 is supported on iOS: https://support.apple.com/en-us/116944
Also safari on iOS seems to support the playback of h265. https://caniuse.com/hevc
Correct me if I am wrong, but WebKit is used to render expo?

**Comment by michi7801** [comment]
@fredrikburmester First of all: Thank you for looking into this issue so quickly.

Your findings made me retest my previous statements about what's working and what's not:
- If I playback from the official Jellyfin App the stream is remuxed into another container (which is weird because I store everything in mp4 containers which are supported according to the docs), but it's NOT transcoded.
- If I playback from SwiftFin it is direct playing without remux or transcoding
- If I playback from StreamyFin only the "old" profile it is working which then consequently transcodes the file to h264
- All other profiles in StreamyFin are not working. They just show the Quicktime splash screen as described.

I love the look and feel of StreamyFin but my poor Synology can't handle that for multiple users.

**Comment by fredrikburmester** [comment]
Yeah Swiftfin uses VLC player which supports many more formats/codecs than the default iOS video player.

**Comment by michi7801** [comment]
Anything I can do to help resolve this issue? It hurts that I can't use this gorgeous app.

With TV support upcoming and the current pace of development this could really get the default app for Jellyfin

**Comment by fredrikburmester** [comment]
A great way to start is to look into why some video files don't get transcoded into a supported container. It's probably because of the device profile doesn't tell the server correctly what codecs/containers are supported. But it might be some other issues.

**Comment by michi7801** [comment]
Yes, I actually caught a difference in behavior. Thanks for the hint!
You initially said that you use the same expo profile as the official Jellyfin client (and I also verified that), but the Jellyfin client actually only does direct stream (it changes the container format) while Streamyfin attempts to direct play. 

This would lead me to the assumption that Streamyfin incorrectly informs that server about the clients capabilities. Does that help you in any way?

**Comment by fredrikburmester** [comment]
You say that Streamyfin Direct plays while JF Direct streams, what do you base that on?

**Comment by michi7801** [comment]
The Admin interface tells you all active streams and their properties. For Jellyfin-Expo it remuxes the container while for Streamyfin it direct plays.

**Comment by fredrikburmester** [comment]
I was afraid you were gonna say that. Unfortunately Streamyfin does not accurately report playback information to the server. Best way to determine what is going on is by looking at the server logs.

**Comment by michi7801** [comment]
I further investigated and found pretty unreliable behavior actually. It turns out that some of my media is remuxed and does playback on streamyfin while other doesn't. This is weird because all my media is converted to a single format through an app called unmanic (essentially an ffmpeg wrapper). So everything I own is h265, SDR, AAC, mp4...

The app actually knows about this before requesting the stream because it already tells you ahead of time "Direct Stream" or "Transcoded Stream". All videos marked with "Transcoded Stream" do playback correctly (although the loading icon does not vanish)

Any idea what is causing that?

**Comment by fredrikburmester** [comment]
The only thing you didn't mention was subtitles. That could definitely cause a transcode and or remux which could cause problems. All of this probably stems from an incorrect device profile, which we are working on.

**Comment by fredrikburmester** [comment]
Also we're very close to release a new integrated VLC player for iOS which will fix all of these problems.

**Comment by michi7801** [comment]
Correct, I do not modify the original subtitles. 
Got it, thank you for helping me investigate this problem. Looking forward to future releases!

**Comment by michi7801** [comment]
Yes, videos with no subtitles at all seem to be the one who direct stream

**Comment by michi7801** [comment]
I worked through my whole library (about 130 Files) and literally the only ones not working are those without any subtitles at all
